src/char_ref.rl

```diff
@@ -164,8 +164,15 @@ static bool consume_numeric_ref(
 
   int codepoint = 0;
   bool status = true;
+  bool poisoned = false;
   do {
-    codepoint = (codepoint * (is_hex ? 16 : 10)) + digit;
+    if (!poisoned) {
+      codepoint = (codepoint * (is_hex ? 16 : 10)) + digit;
+      if (codepoint < 0) {
```

Comment:
This check seems to rely on wrapping overflow semantics but according to the C standard, signed integer overflow is undefined behaviour.

Reply:
True! However, Rust isn't C. `+` is defined as two's complement wrapping when debug mode is disabled.  This will cause issues in debug mode, though, so I'll update it to use `wrapping_add` explicitly. Thanks!

Reply:
wait, this is C, not rust. my GOODNESs, I am not awake today.

Reply:
(I saw `.rl` and thought `.rs`. My apologies.)